wallet.go

```diff
@@ -0,0 +1,144 @@
+package wallet
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"golang.org/x/crypto/sha3"
+)
+
+type ChannelStatus uint8
+
+const (
+	ChannelStatusCooperative ChannelStatus = iota
+	ChannelStatusChallenge
+	ChannelStatusFinalized
+	ChannelStatusIndependent
+)
+
+type Participant uint8
+
+const (
+	ParticipantOwner Participant = iota
+	ParticipantIntermediary
+)
+
+const CurrentBlockNumber = 0 // todo: replace with a service
+
+var myAddress common.Address = common.Address{} // todo: compute from injected private key - cli? config file?
+
+// ChannelWallet is the off-chain accounting struct for a state channel wallet.
+type ChannelWallet struct {
+	//////////////////
+	// Channel Info //
+	//////////////////
+
+	ChannelAddress common.Address
```

Comment:
Would this be the address of the `owner`'s SCW?

Reply:
I think so, yes. It assumes the place of ChannelID in nitro - an identifier for the bundled and signed channel state data